### PR TITLE
Remove ringpop use from integration tests

### PR DESCRIPTION
**Pull request by shreyassrivatsan**
- Create membership factory and hide ringpop initialization behind that
- Refactor prod to use the new membership factory
- Create a simple resolver/monitor for integration tests that works with a static configuration

### CODE REVIEW

File: common/membership/simpleMonitor.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package membership
+
+import "fmt"
+
+type simpleMonitor struct {
```
Comment (venkat1109) [inline common/membership/simpleMonitor.go]: [NOT A BLOCKER] given that this is only used for integration test, how do you feel about moving this code inside the host/ package ?

Comment (shreyassrivatsan) [reply common/membership/simpleMonitor.go]: makes sense. I'll do that..

File: common/service/service.go [hunk 2]
```diff
@@ -76,10 +75,12 @@ type (
...
-	// RingpopFactory provides a bootstrapped ringpop
-	RingpopFactory interface {
-		// CreateRingpop vends a bootstrapped ringpop object
-		CreateRingpop(d *yarpc.Dispatcher) (*ringpop.Ringpop, error)
+	// MembershipFactory provides a bootstrapped membership monitor
+	MembershipFactory interface {
+		// CreateMembershipMonitor vends a bootstrapped membership monitor
+		CreateMembershipMonitor(d *yarpc.Dispatcher) (membership.Monitor, error)
+
+		DestroyMembershipMonitor()
```
[the remark this answers is not in the document]
Comment (shreyassrivatsan) [reply common/service/service.go]: Yes, there is a close on the monitor - but if the ringpop object is created outside it and we destroy it within the monitor, that's odd. I could limit only one call to start for the ringpop monitor.. Let me see if I can structure the code better.

File: common/service/service.go [hunk 3]
```diff
@@ -76,10 +75,12 @@ type (
 		DCRedirectionPolicy config.DCRedirectionPolicy
 	}
 
-	// RingpopFactory provides a bootstrapped ringpop
-	RingpopFactory interface {
-		// CreateRingpop vends a bootstrapped ringpop object
-		CreateRingpop(d *yarpc.Dispatcher) (*ringpop.Ringpop, error)
+	// MembershipFactory provides a bootstrapped membership monitor
+	MembershipFactory interface {
```
[the remark this answers is not in the document]
Comment (shreyassrivatsan) [reply common/service/service.go]: done

**Pull request merged**